clients/admin-ui/src/features/system/SystemFormTabs.tsx

```diff
@@ -109,30 +109,33 @@ const SystemFormTabs = ({
     }
   }, [activeSystem]);
 
-  const handleSuccess = (system: SystemResponse) => {
-    // show a save message if this is the first time the system was saved
-    if (activeSystem === undefined) {
-      setShowSaveMessage(true);
-    }
-    dispatch(setActiveSystem(system));
-    const toastParams = {
-      ...DEFAULT_TOAST_PARAMS,
-      description: (
-        <ToastMessage
-          onViewDatamap={() => {
-            router.push(systemOrDatamapRoute).then(() => {
+  const handleSuccess = useCallback(
```

Comment:
wrapped some things in a `useCallback` while I was debugging why the test was making so many calls. this wasn't the solution, but is good to have regardless